battle.h

```diff
@@ -48,19 +51,14 @@ enum {
 	BATTLE_DRAW
 };
 
-class BattlePtr : public AListElem
-{
-	public:
-		Battle * ptr;
-};
-
```

Comment:
the battle list on both game and factions are now std::vectors that take the Battle * directly.  No more need for this BattlePtr thingy.
Eventually everything that is an AList and derives from AListElem will be std::vector and the heavens will open up and the angels will sing hosannahs to our names.

Reply:
Vectors (or any other collection from std) is a way to go almost everywhere.

Reply:
Yup.. I am converting pretty much all alists and astrings to std:: types as I go.. once I am done with json reports (and fixing the QM quandry) I will go through and do a pass and clear out the rest of the uses of those two and nuke them from orbit :p